pkg/plugins/builtin/script/script.go

```diff
@@ -0,0 +1,142 @@
+package script
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	gexec "os/exec"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/ovh/utask"
+
+	"github.com/ovh/utask/pkg/plugins/taskplugin"
+)
+
+// the script plugin execute scripts
+var (
+	Plugin = taskplugin.New("script", "0.1", exec,
+		taskplugin.WithConfig(validConfig, Config{}),
+	)
+)
+
+// Metadata represents the metadata of script execution
+type Metadata struct {
+	ExitCode      string `json:"exit_code"`
+	ProcessState  string `json:"process_state"`
+	Output        string `json:"output"`
+	ExecutionTime string `json:"execution_time"`
+	Error         string `json:"error"`
+}
+
+// Config is the configuration needed to execute a script
+type Config struct {
+	File    string   `json:"file,required"`
+	Argv    []string `json:"argv,omitempty"`
+	Timeout string   `json:"timeout,omitempty"`
+}
+
+func validConfig(config interface{}) error {
+	cfg := config.(*Config)
+
+	if cfg.File == "" {
+		return errors.New("file is missing")
+
+		f, err := os.Stat(filepath.Join(utask.FScriptsFolder, cfg.File))
+		if os.IsNotExist(err) {
+			return fmt.Errorf("%s not found in FS: %s", cfg.File, err.Error())
+		}
+
+		if f.Mode()&0010 == 0 {
+			return fmt.Errorf("%s haven't exec permissions", cfg.File)
+		}
+	}
+
+	if cfg.Timeout != "" {
+		if cfg.Timeout[0] == '-' {
+			return errors.New("timeout must be positive")
+		}
+		if _, err := time.ParseDuration(cfg.Timeout); err != nil {
+			return fmt.Errorf("timeout is wrong %s", err.Error())
+		}
+	}
+
+	return nil
+}
+
+func exec(stepName string, config interface{}, ctx interface{}) (interface{}, interface{}, error) {
+	cfg := config.(*Config)
+
+	var timeout time.Duration
+
+	if cfg.Timeout != "" {
+		timeout, _ = time.ParseDuration(cfg.Timeout)
+	} else {
+		// default is 2 * 1 minute = 2 minutes
+		timeout = time.Duration(2) * time.Minute
+	}
+
+	ctxe, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	cmd := gexec.CommandContext(ctxe, filepath.Join(utask.FScriptsFolder, cfg.File), cfg.Argv...)
```

Comment:
What about the working directory of the command ?
If you set if, it will inherit from the working directory of the parent process.
IMO we should set it to the directory of `filepath.Join(utask.FScriptsFolder, cfg.File)`. It makes the behavior more predictable for a script developer.

Reply:
Fixed.